pkg/processor/test/suite/suite.go

```diff
@@ -284,7 +302,18 @@ func (suite *TestSuite) PopulateDeployOptions(createFunctionOptions *platform.Cr
 }
 
 func (suite *TestSuite) GetUniqueFunctionName(name string) string {
-	return fmt.Sprintf("%s-%s", name, suite.TestID)
+	uniqueFunctionName := fmt.Sprintf("%s-%s", name, suite.TestID)
```

Comment:
That's a misleading name, and this tactic can easily break - it merely concats the name to the TestID and hopes the same TestID won't call this twice with the same name.
So, unique name is still achieved by a distributed effort of methods.
This holds water for now but looks like this should, to be reliable, truly randomize part of the name